src/main/java/me/botsko/prism/actions/BlockAction.java

```diff
@@ -30,574 +30,534 @@
+                    Prism.debug(parameters.getProcessType().name() + " skipped due to lack of wrong sibling type for "
+                            + getMaterial().name());
+                    return new ChangeResult(ChangeResultType.SKIPPED, null);
+                }
+
+                sibling.setType(block.getType());
+
+                BlockData siblingData = getBlockData().clone();
+
+                if (siblingData instanceof Bed) {
+                    // We always log the foot
+                    ((Bed) siblingData).setPart(Part.HEAD);
+                } else if (siblingData instanceof Bisected) {
+                    // We always log the bottom
+                    ((Bisected) siblingData).setHalf(Half.TOP);
+                }
+
+                sibling.setBlockData(siblingData);
+            }
+        }
+
+        boolean physics = !parameters.hasFlag(Flag.NO_PHYS);
+
+        state.update(true, physics);
+
+        if (sibling != null) {
+            sibling.update(true, physics);
+        }
+        return null;
+    }
+
+    private ChangeResult handleSkulls(final Block block, BlockActionData blockActionData,
+                                      final BlockState originalBlock) {
+        block.setType(getMaterial());
+        BlockState state = block.getState();
+        final SkullActionData s = (SkullActionData) blockActionData;
+
+        if (state.getBlockData() instanceof Rotatable) {
+            final Rotatable r = (Rotatable) state.getBlockData();
+            r.setRotation(s.getRotation());
+            state.setBlockData(r);
+        } else {
+            final Directional d = (Directional) state.getBlockData();
+            d.setFacing(s.getRotation());
+            state.setBlockData(d);
+        }
+        state = block.getState();
+
+        if (!s.owner.isEmpty()) {
+            final Skull skull = (Skull) state;
+            skull.setOwningPlayer(Bukkit.getOfflinePlayer(EntityUtils.uuidOf((s.owner))));
+        }
+        BlockStateChange stateChange = new BlockStateChange(originalBlock, state);
+        return new ChangeResult(ChangeResultType.APPLIED, stateChange);
+    }
+
+    ChangeResult removeBlock(Player player, QueryParameters parameters, boolean isPreview, Block block) {
+
+        BlockStateChange stateChange;
+
+        if (!block.getType().equals(Material.AIR)) {
+
+            // Ensure it's acceptable to remove the current block
+            if (!BlockUtils.isAcceptableForBlockPlace(block.getType())
+                    && !BlockUtils.areBlockIdsSameCoreItem(block.getType(), getMaterial())
+                    && !parameters.hasFlag(Flag.OVERWRITE)) {
+                return new ChangeResult(ChangeResultType.SKIPPED, null);
+            }
+
+            if (!isPreview) {
+
+                // Capture the block before we change it
+                final BlockState originalBlock = block.getState();
+
+                // Set
+                block.setType(Material.AIR);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Material.AIR' due to existing static import 'org.bukkit.Material.*'](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5446892)